sketch_map_tool/upload_processing/qr_code_reader.py

```diff
@@ -62,7 +62,7 @@ def _decode_data(data) -> MappingProxyType:
             *[float(coordinate) for coordinate in contents[2:-1]]
         )  # Raises ValueError for non-float values
         try:
-            layer = getattr(Layer, (contents[7]))
+            layer = getattr(Layer, (contents[6]).upper().replace("-", "_"))
         except IndexError:
             # backward compatibility
             layer = getattr(Layer, "OSM")
```

Comment:
```suggestion
            layer = Layer(contents[6])
        except IndexError:
            # backward compatibility
            layer = Layer("osm")
```

Would that also work? You would need to try it out.

Reply:
done